docs/DeveloperGuide.md

```diff
@@ -1,38 +1,254 @@
+the `DepositCommand` class
+will run, creating a new `DepositAction` class. The `execute` function will run a method corresponding to what the user
+inputs. The design
+of this three class system is meant to modularize the different aspects of the internal logic so future problems would
+be encapsulated in a specific location.
 
-## Instructions for manual testing
+### Expense Component
+
+### ExpenseCommand Class
+
+The `ExpenseCommand` class contains methods that relate to the execution of the expense functionality in BudgetBuddy.
+Users can
+create new expense entries with a timestamp (if necessary). These expenses are then stored in an expense list and users
+can delete existing
+expenses, find expenses using keywords, and list all expenses according to their categories.
+
+The user input is split by the parser in the `CommandParser` class and is redirected to the `ExpenseCommand` class if
+the first word is
+"expense". Based on the second command word, a method in `CommandAction` class is called corresponding to the command
+entered by the user.
+
+Attached below is how the `ExpenseCommand` class is implemented along with its relation with the other `Data` classes
+and the abstract
+`Command` class.
+
+![ExpenseCommand.png](uml/ExpenseCommand.png)
+
+### ExpenseCommand Sequence
+
+Once a string `input` from the user has been deemed as an `Expense` command, the `input` will be passed into the
+`ExpenseCommand` class. This class determines what expense commands to carry out to update the `data` stored in the
+`Budget` and `Expense` array lists, as shown in the sequence diagram below.
+
+![img.png](uml/ExpenseCommandSequence.png)
+
+### Design & Implementation of the Expense Feature
+
+The expense feature is similarly modularized and designed with an OOP lens. As such, its functionalities have been
+separated into different
+classes such as the parsing of user input, and the parsing and execution of each of its features.
+
+On a high level, the expense feature starts with `CommandParser` taking in the input and choosing which `Command` class
+to execute from.
+This happens with all user input in BudgetBuddy. Then, if the first word is "expense", the `execute` function of
+the `ExpenseCommand` class
+will run, creating a new `ExpenseAction` class. The `execute` function will run a method corresponding to what the user
+inputs. The design
+of this three class system is meant to modularize the different aspects of the internal logic so future problems would
+be encapsulated in
+a specific location.
+
+### Stats Component
+
+### StatsCommand Class
+
+The 'StatsCommand' class contains methods that are related to the execution of the stats functionality in BudgetBuddy.
+Users can
+use this command to view all the details of their expenses, budgets and deposits in the current month. Through this
+feature, users will know
+their current progress and if their expenses have exceeded their budget. 
+
+![StatsCommand.png](uml/StatsCommand.png)
```

Comment:
agrred with vishnuvk47, good layout of the diagram. I think 0... may be replaced with * instead?